### PR TITLE
New package: fcron-3.2.0

### PR DESCRIPTION
**Pull request by thypon**
#1672

### CODE REVIEW

File: srcpkgs/fcron/INSTALL [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	chown root:fcron /usr/bin/fcronsighup
+
+	for _f in /etc/fcron/fcron.{conf,allow,deny}; do
+		[[ -e $_f ]] || continue
+		chown root:fcron "$_f"
+		chmod 640 "$_f"
+	done
+
+	chown fcron:fcron /var/spool/fcron
+	chmod 770 /var/spool/fcron
```
Comment (beefcurtains) [inline srcpkgs/fcron/INSTALL]: Why not do all of this in post_install()?

File: srcpkgs/fcron/INSTALL [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+post)
+	# fix permissions and owners
+	chmod  110 /usr/bin/fcron
+	chmod 4110 /usr/bin/fcronsighup
+	chmod 6111 /usr/bin/fcrontab /usr/bin/fcrondyn
+	chown fcron:fcron /usr/bin/fcrontab /usr/bin/fcrondyn
+	chown root:fcron /usr/bin/fcronsighup
+
+	for _f in /etc/fcron/fcron.{conf,allow,deny}; do
+		[[ -e $_f ]] || continue
```
Comment (xtraeme) [inline srcpkgs/fcron/INSTALL]: INSTALL/REMOVE scripts are meant to be run by POSIX shell... please remove those bashisms.

File: srcpkgs/fcron/template [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	/etc/fcron/fcron.deny
+	/var/spool/fcron/systab.orig"
+
+## me: fcron installer is not the king. I'm.
+## fcron: Wine is coming.
+pre_configure() {
+	touch /usr/bin/vi /usr/bin/sendmail
+	chmod 755 /usr/bin/vi /usr/bin/sendmail
+	echo "fcron:x:23:" >> /etc/group
+	echo "fcron:fcron:23:23:FCron User:/var/spool/fcron:/bin/sh" >> /etc/passwd
```
Comment (beefcurtains) [inline srcpkgs/fcron/template]: Radical! I'll look into patching the configure.ac instead... :smile::hammer:

File: srcpkgs/fcron/INSTALL (hunk 1 above)
Comment (thypon) [inline srcpkgs/fcron/INSTALL]: seems that using chgrp,chown launch an error. I prefer to mantain these fixes in the same place.

File: srcpkgs/fcron/template [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	--sysconfdir=/etc/fcron
+	--with-answer-all=no
+	--with-boot-install=no
+	--datarootdir=/usr/share
+	--datadir=/usr/share
+	--docdir=/usr/share/doc
+	--localstatedir=/var
+	--with-piddir=/run
+	--with-sendmail=/usr/bin/sendmail
+	--with-editor=/usr/bin/vi
```
Comment (beefcurtains) [inline srcpkgs/fcron/template]: Just ``sendmail`` and ``vi`` here, please. And drop the ``nvi`` depends.

File: srcpkgs/fcron/template [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	/etc/fcron/fcron.conf
+	/etc/fcron/fcron.allow
+	/etc/fcron/fcron.deny
+	/var/spool/fcron/systab.orig"
+
+## me: fcron installer is not the king. I'm.
+## fcron: Wine is coming.
+pre_configure() {
+	touch /usr/bin/vi /usr/bin/sendmail
+	chmod 755 /usr/bin/vi /usr/bin/sendmail
```
Comment (beefcurtains) [inline srcpkgs/fcron/template]: ``vi`` is not needed due to the ``--disable-checks`` argument above. Sendmail could be added as a host dep, or patched out be removing the relevant ``AC_MSG_ERROR`` line in ``configure.in``

File: srcpkgs/fcron/template (hunk 4 above)
Comment (thypon) [inline srcpkgs/fcron/template]: A vi implementation is in the unix spec. Right now nvi is also included in hour base-system. If you want to use another you could use EDITOR env variable.

**Comment by beefcurtains** [comment]
[Here's a patch](https://gist.github.com/beefcurtains/81940a5b0176abc08c8a), testing now...

**Comment by thypon** [comment]
the pre_install part is necessary to pass the make phase.

**Comment by beefcurtains** [comment]
That was just a test to see what happens. The ``chmod``s should be done in ``post_install``, is there any particular reason for doing it in ``INSTALL``?

**Comment by thypon** [comment]
Yep, it's a xbps-src workaround. As you see the perms are -r. However without read permission xbps-src can't create the hashes to construct the package. The `make install` leave, more or less, these permission on files but before packaging i need to leave +r, so i fix this thing on extraction.

**Comment by beefcurtains** [comment]
There's no reason to remove the read bit after installation, the Makefile is just being stupid here.

**Comment by thypon** [comment]
Yep and not. 3 out of 4 executable(s) are suid. Being not readable leave no possibility to an attacker that wants to create a ropchain to see the application layout without other information disclousure vector.

**Comment by beefcurtains** [comment]
That's stupid, an attacker can just download the xbps file and take a look.

**Comment by thypon** [comment]
True. Not so usefull in a rolling release + not knowing if the package is custom built or updated. That argument is less more valid in debian/ubuntu/fedora.

**Comment by beefcurtains** [comment]
None of the other setuid binaries are -r, and no one has a problem with that. This reeks of security through obscurity.

**Comment by thypon** [comment]
It's a little overkill. But it's exactly the reason i want this priviledge separated cron server in voidlinux. It's obvious it's not a great security feature if you run the same version in current repository, but it's quite good to avoid run anywhere attacks where a script tries to read the application layout + tries to find gadget to use rops and avoid ASLR.

**Comment by xtraeme** [comment]
You can't run chown/chgrp in templates because xbps-src runs as a regular user. Permissions must be set thru INSTALL scripts.

File: srcpkgs/fcron/INSTALL [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# *-*-shell-*-*
+#
+case ${ACTION} in
+post)
+	# fix permissions and owners
+	chown fcron:fcron /usr/bin/fcrontab /usr/bin/fcrondyn
```
Comment (xtraeme) [inline srcpkgs/fcron/INSTALL]: Do not use absolute paths, use relative... i.e `chown ... usr/bin/foo`

File: srcpkgs/fcron/template [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+makedepends="pam-devel readline-devel"
+depends="run-parts"
+short_desc="Feature-rich cron implementation"
+maintainer="Andrea Brancaleoni <miwaxe@gmail.com>"
+license="GPL-2"
+homepage="http://fcron.free.fr"
+distfiles="$homepage/archives/$pkgname-$version.src.tar.gz"
+checksum=eb13b86ed96805d117e19847f1da30b834f1990f7ef4c38f5007234d44679564
+
+system_accounts="fcron:23"
```
Comment (xtraeme) [inline srcpkgs/fcron/template]: Why this needs to set `uid 23` specifically?

File: srcpkgs/fcron/template [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+fcron_homedir="/var/spool/fcron"
+fcron_shell="/bin/sh"
+
+conf_files="
+	/etc/fcron/fcron.conf
+	/etc/fcron/fcron.allow
+	/etc/fcron/fcron.deny
+	/var/spool/fcron/systab.orig"
```
Comment (xtraeme) [inline srcpkgs/fcron/template]: PAM files are missing from `conf_files`.

**Comment by xtraeme** [comment]
Looks ok :+1:

**Pull request merged**